test/test_type_inference.py

```diff
@@ -0,0 +1,20 @@
+from acclimatise.cli_types import CliString
+from acclimatise.model import infer_type
+
+
+def test_type_inference():
+    test_strings = [("", CliString, False), ("int", CliInteger, False),
```

Comment:
This is a great array of test cases, but please do this using more idiomatic pytest, e.g. https://docs.pytest.org/en/latest/parametrize.html#pytest-mark-parametrize-parametrizing-test-functions